src/connector/src/sink/encoder/json.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use base64::engine::general_purpose;
+use base64::Engine as _;
+use chrono::{Datelike, Timelike};
+use risingwave_common::array::{ArrayError, ArrayResult};
+use risingwave_common::row::Row;
+use risingwave_common::types::{DataType, DatumRef, ScalarRefImpl, ToText};
+use risingwave_common::util::iter_util::ZipEqDebug;
+use serde_json::{json, Map, Value};
+
+use super::{Field, Result, RowEncoder, RowRef, SerTo, TimestampHandlingMode};
+use crate::sink::SinkError;
+
+pub struct JsonEncoder {
+    timestamp_handling_mode: TimestampHandlingMode,
+}
+
+impl JsonEncoder {
+    pub fn new(timestamp_handling_mode: TimestampHandlingMode) -> Self {
+        Self {
+            timestamp_handling_mode,
+        }
+    }
+}
+
+impl RowEncoder for JsonEncoder {
+    type Output = Map<String, Value>;
+
+    fn encode(
+        &self,
+        row: RowRef<'_>,
+        schema: &[Field],
+        col_indices: impl Iterator<Item = usize>,
+    ) -> Result<Self::Output> {
+        let mut mappings = Map::with_capacity(schema.len());
+        for idx in col_indices {
+            let field = &schema[idx];
+            let key = field.name.clone();
+            let value =
+                datum_to_json_object(field, row.datum_at(idx), self.timestamp_handling_mode)
+                    .map_err(|e| SinkError::JsonParse(e.to_string()))?;
+            mappings.insert(key, value);
+        }
+        Ok(mappings)
+    }
+}
+
+impl SerTo<String> for Map<String, Value> {
+    fn ser_to(self) -> Result<String> {
+        Value::Object(self).ser_to()
+    }
+}
+
+impl SerTo<String> for Value {
+    fn ser_to(self) -> Result<String> {
+        Ok(self.to_string())
+    }
+}
+
+fn datum_to_json_object(
```

Comment:
All lines below (i.e. `datum_to_json_object` and `mod tests`) are moved from `sink::utils` without changing a single character.